components/nexusoperations/frontend/handler.go

```diff
@@ -524,3 +528,29 @@ func (c *requestContext) shouldForwardRequest(ctx context.Context, header http.H
 		len(c.namespace.ClusterNames()) > 1 &&
 		c.Config.ForwardingEnabledForNamespace(c.namespace.Name().String())
 }
+
+// enrichNexusOperationMetrics enhances metrics with additional Nexus operation context based on configuration.
+func (c *requestContext) enrichNexusOperationMetrics(service, operation string, requestHeader http.Header) {
+	conf := c.metricTagConfig.Get()
+	if conf == nil {
+		return
+	}
+
+	var tags []metrics.Tag
+
+	if conf.IncludeServiceTag {
+		tags = append(tags, metrics.NexusServiceTag(service))
+	}
+
+	if conf.IncludeOperationTag {
+		tags = append(tags, metrics.NexusOperationTag(operation))
+	}
+
+	for _, mapping := range conf.HeaderTagMappings {
```

Comment:
This config was meant for the handler side, I'm a bit hesitant to also apply it on the caller side but I guess there's no harm in doing so.